cis_policies_generator/src/index.ts

```diff
@@ -0,0 +1,173 @@
+import _ from "underscore";
+import fs from 'fs';
+import config from 'config';
+import xlsx from 'node-xlsx';
+import path from 'path';
+// import axios from 'axios';
+import Promise from "bluebird";
+import YAML from 'yaml';
+
+const {v5: uuidv5} = require('uuid');
+
+const output_folder: string = config.get("output_folder");
+const benchmarks_folder: string = config.get("benchmarks_folder");
+
+function generateOutputFolder(): void {
+    console.log("Creating output folder:", output_folder);
+    fs.rmdirSync(output_folder, {recursive: true});
+    fs.mkdirSync(output_folder);
+}
+
+function parseReferences(references): Promise<string[]> {
+    if (!references) {
+        return Promise.resolve([]);
+    }
+
+    const links: string[] = references.replaceAll(":http", "\nhttp").split("\n");
+    return Promise.resolve(links);
+}
+
+// Generate a mapping between section # and its title.
+// For example: { 'section #': '1', title: 'Control Plane Components' }
+function parseAllSections(data): SectionMetadata[] {
+    let arr = _.filter(data, (el) => {
+        return (el["section #"] && el["title"] && !el["recommendation #"]);
+    });
+    arr = _.map(arr, i => _.pick(i, ["section #", "title"]));
+    return arr;
+}
+
+function identifySection(rule_section: string, sections: SectionMetadata[]): string {
+    for (const section of sections) {
+        if (section["section #"] == rule_section) {
+            return section.title;
+        }
+    }
+    // we should never get here!
+    console.log("FATAL: Could not find section for rule", rule_section);
+    process.exit(-1);
+}
+
+// Adds newline character after code-blocks that are at the end of a property
+function fixCodeBlocks(val: string): string {
+    if (val.endsWith("```")) {
+        return val.concat("\r\n");
+    }
+    return val;
+}
+
+function normalizeResults(data: BenchmarksData[], benchmark_metadata: BenchmarkMetadata,
+                          profile_applicability: string): Promise<RuleSchema[]> {
+    const sections = parseAllSections(data);
+
+    let result = _.filter(data, (it) => {
+        return Boolean(it["recommendation #"]);
+    });
+
+    return Promise.map(result, (it) => {
```

Comment:
I'm not quite sure promises are really needed in this use case, we do not perform any async operation.

Reply:
true